backend/core/src/listings/listings.service.ts

```diff
@@ -101,18 +134,25 @@ export class ListingsService {
     }
     const view = getView(this.listingRepository.createQueryBuilder("listings"), params.view)
 
-    let listings = await view
+    let mainQuery = view
       .getViewQb()
       .andWhere("listings.id IN (" + innerFilteredQuery.getQuery() + ")")
       // Set the inner WHERE params on the outer query, as noted in the TypeORM docs.
       // (WHERE params are the values passed to andWhere() that TypeORM escapes
       // and substitues for the `:paramName` placeholders in the WHERE clause.)
       .setParameters(innerFilteredQuery.getParameters())
-      .orderBy(ListingsService.getOrderByCondition(params))
-      // Order by units.maxOccupancy is applied last so that it affects the order
-      // of units _within_ a listing, rather than the overall listing order)
-      .addOrderBy("units.max_occupancy", "ASC", "NULLS LAST")
-      .getMany()
+
+    for (const orderByCondition of orderByConditions) {
+      mainQuery = mainQuery.addOrderBy(
+        orderByCondition.key,
+        orderByCondition.order,
+        orderByCondition.nulls
+      )
+    }
```

Comment:
if we wait to add the `andWhere` and `setParameters` to `mainQuery`, can we loop over orderByConditions once and set the orderBy to `mainQuery` and `innerFilteredQuery` at the same time?